session.go

```diff
@@ -698,9 +702,22 @@ func (q *Query) defaultsFromSession() {
 	q.serialCons = s.cfg.SerialConsistency
 	q.defaultTimestamp = s.cfg.DefaultTimestamp
 	q.idempotent = s.cfg.DefaultIdempotence
+	q.metrics = initializeMetrics(s)
 	s.mu.RUnlock()
 }
 
+func initializeMetrics(s *Session) map[string]*queryMetric {
```

Comment:
the main reason is that Query and Batch initialisers are the same an don't really need to work on a query object.

On the other hand, acking the comment on the initialiser itself, I will probably get rid of it completely as it would no longer make sense if the nodes are added later.